.github/workflows/ci.yml

```diff
@@ -0,0 +1,47 @@
+name: CI
+on:
+  push:
+    branches-ignore: [ main, master ] # ignoring branch when will be pushed using release build TODO: use main branch in next release
+    paths-ignore:
+      - ".github/**"
+      - "**.md"
+      - ".gitignore"
+      - ".editorconfig"
+  pull_request:
+    branches: [ develop ]
+    paths-ignore:
+      - ".github/**"
+      - "**.md"
+      - ".gitignore"
+      - ".editorconfig"
+
+env:
+  RELEASE: false
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I don't think the GH TOKEN is needed here - also we can't use any secrets in workflows that run for pull requests.

Reply:
This should be not needed in test unit. But if you push something into gh pages, registry or the release